website_snippet_mass_mailing_partner/static/src/js/website_snippet_mass_mailing_partner.js

```diff
@@ -0,0 +1,57 @@
+/* © 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+"use strict";
+(function ($) {
+    openerp.website.snippet.animationRegistry.subscribe.include({
+        start: function(editable_mode) {
+            var self = this;
+            self._super(editable_mode);
+            if (self.$target.find(".js_subscribe_name:visible")) {
+                openerp.jsonRpc('/website_mass_mailing/is_subscriber', 'call', {
+                    list_id: this.$target.data('list-id'),
+                }).then(function (data) {
+                    self.$target.find('.js_subscribe_name:visible')
+                    .val(data.name)
+                    .attr(
+                        "disabled",
+                        Boolean(data.is_subscriber && data.name.length)
+                    );
+                });
+            }
+        },
+        on_click: function() {
```

Comment:
be creative. You can monkey patch `openerp.jsonRpc` to interject a call to ``/website_mass_mailing/subscribe`` and add a success handler. Or just inject your value then. Or use http://api.jquery.com/ajaxStop. Or write the name into the session before you call super and get this value in your controller